blink/hostfs.c

```diff
@@ -128,6 +128,21 @@ int HostfsInit(const char *source, u64 flags, const void *data,
   return -1;
 }
 
+int HostfsReadmountentry(struct VfsDevice *device, char **spec, char **type, char **mntops) {
+  struct HostfsDevice *hostfsdevice = (struct HostfsDevice *)device->data;
+  *spec = strdup(hostfsdevice->source);
```

Comment:
One best practice for API design in C is that the output parameters should only be modified on success. There's very few exceptions to this rule. But it's mostly an issue for public APIs.